### ISSUE TITLE
Crash on centos 7.3 & mono 5 when doing a parallel test

### ISSUE THREAD

**Issue by wavebit** [akkadotnet/akka.net#2898]
Hello,
I modified the chat example to do a parallel testing.
This code is I added.

```csharp
Parallel.For(0, 1000, i =>
{
    {
        chatClient.Tell(new SayRequest()
        {
            Text = input + " " + i.ToString(),
            Seq = i
        });
    }
});
```

The server-side works fine on windows, but carsh on Centos 7.3 and mono 5.

The exception I got:

> Unhandled Exception:
System.ArgumentException: Buffer is empty
Parameter name: buffers
  at System.Net.Sockets.SocketAsyncResult.CheckIfThrowDelayedException () [0x00014] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult asyncResult, System.Net.Sockets.SocketError& errorCode) [0x00055] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult asyncResult) [0x00000] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket+<>c.<.cctor>b__306_12 (System.IAsyncResult ares) [0x0002c] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.SocketAsyncResult+<>c__DisplayClass27_0.<Complete>b__0 (System.Object _) [0x00000] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00008] in <3753d1715b8842d8bb13a30db0388b60>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <3753d1715b8842d8bb13a30db0388b60>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <3753d1715b8842d8bb13a30db0388b60>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: Buffer is empty
Parameter name: buffers
  at System.Net.Sockets.SocketAsyncResult.CheckIfThrowDelayedException () [0x00014] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult asyncResult, System.Net.Sockets.SocketError& errorCode) [0x00055] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket.EndSend (System.IAsyncResult asyncResult) [0x00000] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.Socket+<>c.<.cctor>b__306_12 (System.IAsyncResult ares) [0x0002c] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Net.Sockets.SocketAsyncResult+<>c__DisplayClass27_0.<Complete>b__0 (System.Object _) [0x00000] in <1d7393d853954016b607e8e348b00ad1>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00008] in <3753d1715b8842d8bb13a30db0388b60>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <3753d1715b8842d8bb13a30db0388b60>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <3753d1715b8842d8bb13a30db0388b60>:0

**Comment by Horusiath** [comment]
@wavebit is this a whole stack trace? It looks like some parts are missing. Also, which Akka.NET version are you using?

**Comment by wavebit** [comment]
yeah, that's a whole stack trace. Both Akka.NET and Akka.Remote version are 1.2.3 from NuGet.
I found that I can reproduce it on a remote server but cann't on a local server in VMware.The lentacy to remote server is about  70ms, to local server in VMware is 0 ms.
![cap](https://user-images.githubusercontent.com/11334491/28523931-1fcbe524-70b1-11e7-8b62-8e6827912b67.PNG)

**Comment by Aaronontheweb** [comment]
I'm scratching my head here a bit... It looks like DotNetty tried to write an empty buffer? I don't see what else could be happening under the covers here

**Comment by aengesse** [comment]
Has any progress been made on this?  I'm experiencing the same issue.

**Comment by Aaronontheweb** [comment]
@aengesse on CentOS?

**Comment by aengesse** [comment]
Yes sir

**Comment by aengesse** [comment]
@Aaronontheweb One thing to note, a similar solution using helios seems to work regularly on the VM containers.  I've only seen this issue with the  dot-netty transport layer configuration.

**Comment by Aaronontheweb** [comment]
@aengesse it has to be an issue with DotNetty itself then and not Akka.NET's transport impl, since this is happening at a layer below us.

**Comment by Aaronontheweb** [comment]
Recommend logging an issue over there

**Comment by aengesse** [comment]
#https://github.com/Azure/DotNetty/issues/279

**Comment by Aaronontheweb** [comment]
Thanks!

**Comment by Aaronontheweb** [comment]
Verified that this is an issue with DotNetty, but we'll resolve it via patch once v0.4.7 of DotNetty is shipped.

**Comment by Aaronontheweb** [comment]
1.3.1 or 1.3.2 depending on timeframe

**Comment by jannicklange** [comment]
Hey!

I stumbled upon the same Issue when trying to start my Application on a Cluster of my university.
Interestingly, the same version of my application runs without any issue on a second cluster.

Bot Clusters (only) provide mono 4.6.2, however they use different versions of Linux:
Failing on
    Linux 2.6.32-504.16.2.el6.x86_64 x86_64
Running on
    Linux 3.10.0-327.36.1.el7.x86_64 x86_64

It seems as though DotNetty released v0.4.7 just a couple of days ago:
https://www.nuget.org/packages/DotNetty.Common/

Can I just update my DotNetty locally, or do I need to wait until you release v1.3.3?
